### DISCUSSION TITLE
How its done?

### DISCUSSION THREAD

**Discussion by quadrixx** [mapsforge/vtm#1078] [Q&A] [answered]
Hello devemux86. I am sorry if this question is foolish, but i really have no clue how to do what i desire. My goal is to make a navigation system in the building. Yes, using floor plans... So, i have a .CAD file of the floor plan. What to do with this file, so mapsforge/vtm may open it. I really am amateur.

**Comment by devemux86** [comment, marked as answer]
If you want to create offline Mapsforge maps (from any source), you can use the map-writer.

Mapsforge map-writer is a plugin for [Osmosis](https://wiki.openstreetmap.org/wiki/Osmosis) software.
You may need to convert your data to a format that Osmosis can read, such as osm or pbf files.
And use tags to define the map elements.

Please see the [map-writer documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md) and the [tag-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml) we use in Mapsforge offline maps.

**Reply by quadrixx** [reply]
waay to go... Thanks for the reply, much appreciated.
